serving/src/main/java/feast/serving/specs/CachedSpecService.java

```diff
@@ -122,7 +122,13 @@ public FeatureSetSpec getFeatureSetSpec(String featureSetRef) throws ExecutionEx
                       generateFeatureStringRef(featureReference), "");
               if (featureSet.isEmpty()) {
```

Comment:
It's strange that we are using `getOrDefault` here and then testing for the default. Isn't that what `get` is for?

Reply:
(not that you wrote it, but possibly in scope)

Reply:
Resolved by replacing `.getOrDefault()` with `.get()` and changing the if condition to check for the `null` returned my `.get()`.